### DISCUSSION TITLE
not able to toggle hide/show line numbers for all views

### DISCUSSION THREAD

**Discussion by jmario74** [orbitalquark/textadept#232] [Q&A] [answered]
needed for small screens; is there a way to toggle hide/show for all views with same keys?

**Comment by orbitalquark** [comment, marked as answer]
You could put the following in your *~/.textadept/init.lua*:

    local margin_widths = {}
    local function toggle_margins()
      for i = 1, view.margins do
        if view.margin_width_n[i] > 0 then
          margin_widths[i] = view.margin_width_n[i]
          view.margin_width_n[i] = 0
        else
          view.margin_width_n[i] = margin_widths[i] or 0
        end
      end
    end

Then bind a key to it like `keys['ctrl+alt+m'] = toggle_margins`
